front50-migrations/src/main/java/com/netflix/spinnaker/front50/migrations/RedBlackToBlueGreenK8sPipelinesMigration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.front50.migrations;
+
+import com.netflix.spinnaker.front50.api.model.pipeline.Pipeline;
+import com.netflix.spinnaker.front50.model.pipeline.PipelineDAO;
+import java.time.Clock;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class RedBlackToBlueGreenK8sPipelinesMigration implements Migration {
+
+  private static final LocalDate VALID_UNTIL = LocalDate.of(2023, 6, 1);
+  private final Clock clock = Clock.systemDefaultZone();
+
+  private final PipelineDAO pipelineDAO;
+
+  public RedBlackToBlueGreenK8sPipelinesMigration(PipelineDAO pipelineDAO) {
+    this.pipelineDAO = pipelineDAO;
+  }
+
+  @Override
+  public boolean isValid() {
+    return LocalDate.from(clock.instant()).isBefore(VALID_UNTIL);
+  }
+
+  @Override
+  public void run() {
+    log.info(
+        "Starting the migration of K8s pipelines using RED/BLACK traffic management strategy to BLUE/GREEN");
+    pipelineDAO.all().stream()
```

Comment:
I am slightly worried about pulling all pipelines from the database and filtering them in memory. Would it make sense to create designated method to get pipelines that match certain criteria? I can imagine something like pipelineDAO.getAll(String provider) to filter out some pipelines on db level?

I am not saying this is the only way to go. Just sharing a tiny concern.

Reply:
I did it like this since this pattern is used in all other places that interact with the `PipelineDao`. Even the helper method `getPipelinesByApplication` is using the same pattern 
```
public Collection<Pipeline> getPipelinesByApplication(String application, boolean refresh) {
    return all(refresh).stream()
        .filter(
            pipeline ->
                pipeline.getApplication() != null
                    && pipeline.getApplication().equalsIgnoreCase(application))
        .collect(Collectors.toList());
  }
```

I agree this is not the best way to do it, I do not like it either but I think this should be the subject for a different PR

Reply:
Actually... this is pipelines.  WE REALLY REALLY don't want PIPELINES In memory.  Those can be LARGE in size.

Reply:
May not be avoidable with todays system.  Just... thinking ideally this would be batched/paginated at least kinda thing.  Nominally